services/bitbucket/bitbucket-pull-request.service.js

```diff
@@ -47,19 +98,33 @@ function pullRequestClassGenerator(raw) {
     static get route() {
       return {
         base: `bitbucket/${routePrefix}`,
-        format: '([^/]+)/([^/]+)',
-        capture: ['user', 'repo'],
+        format: `(?:(http|https)/(.+)/)?([^/]+)/([^/]+)`,
+        capture: ['proto', 'hostAndPath', 'user', 'repo'],
       }
     }
 
     static get examples() {
       return [
         {
           title: 'Bitbucket open pull requests',
-          exampleUrl: 'atlassian/python-bitbucket',
+          namedParams: {
+            user: 'atlassian',
+            repo: 'python-bitbucket',
+          },
           pattern: ':user/:repo',
           staticExample: this.render({ prs: 22 }),
         },
+        {
+          title: 'Bitbucket Server open pull requests',
+          namedParams: {
+            proto: 'https',
+            hostAndPath: 'bitbucket.mydomain.net',
```

Comment:
It sounds like there was some consideration about using a query parameter. Re https://github.com/badges/shields/pull/2598#issuecomment-450236192, I'd like to move away from complicated patterns since they are difficult to express in the UI.

I like the interface that you're using here, though `?server_url=...` would be easier to express in the UI. I wonder if there's any difference in ease of use, from the user's perspective.

I don't have a strong feeling either way, though happy to continue to discuss.

Reply:
I'm also good with query params, just haven't worked with any services like that so I suggested the pattern

Reply:
As of now the param validation has to happen in the handler, though there's no reason it couldn't be added to BaseService, down the line.

Reply:
Given `hostAndPath` is optional, I think it makes more sense for it to be a GET param in this case instead of being part of the route (especially given its optional and in the middle of the route). It makes more sense to think of this case as being analogous to the NPM badges where we're mostly going to be using NPM itself but we can optionally specify a custom registry, as opposed to something like JIRA or Jenkins CI where we always need to manually specify an instance.

Reply:
I'm 👍  on using query params here too, but just a side note/question:

Isn't `hostAndPath` optional here though because the current/existing implementation assumes Bitbucket Cloud as the default instance? The Jira instance could've assumed Jira Cloud as the default instance as well (and then made `hostAndPath` optional) it just didn't make that assumption

Reply:
My only hesitation on using a query param is people who are manually writing badge paths have to URL-Encode the param. I'm not sure how often people would run into that but I think it's still something to consider.

Reply:
Think converting this to a query parameter is the main thing pending on this.